### PR TITLE
Feat/top categories

### PR DESCRIPTION
**Pull request by RebeccaLiu1**
Displays the top 3 spending categories of selected time range in dashboard. 
Screen Shot 2023-08-05 at 8 32 49 PM

NOTE: this uses the `amountInCAD` column from https://github.com/phillytan/FinanceTracker/pull/38.

### CODE REVIEW

File: client/src/components/Dashboard/TopTransactionCategories.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                break;
+        }
+        dispatch(getTopCategoriesAsync({ startDate, endDate }));
+    }, [timeRange])
+
+    // useEffect(() => {
+    //     dispatch(getTopCategoriesAsync())
+    // }, []);
+
+    console.log(topCategories);
```
Comment (vangeliq) [inline client/src/components/Dashboard/TopTransactionCategories.js:57]: these can be deleted i think

File: client/src/redux/services/transactionService.js [hunk 2]
```diff
@@ -5,6 +5,13 @@ const getTransactions = async () => {
   return data;
 };
 
+const getTopCategories = async ({startDate, endDate}) => {
+  console.log("service" + startDate, endDate);
```
Comment (vangeliq) [inline client/src/redux/services/transactionService.js:9]: this one can also be deleted

File: server/routes/transactions.js [hunk 3]
```diff
@@ -2,37 +2,79 @@ var express = require('express');
...
+      return res.status(400).send(error);
+    });
 });
 
+router.get('/topCategories/:startDate/:endDate', function (req, res) {
+  const userId = new mongoose.Types.ObjectId(req.user._id);
+  const startDate = req.params.startDate;
+  const endDate = req.params.endDate;
+
+  console.log(startDate, endDate);
```
Comment (vangeliq) [inline server/routes/transactions.js:26]: console.log statement

**Review by vangeliq — CHANGES_REQUESTED** [review]
I'm not sure if the currencies are converted? but other than that I think it looks good, just some console.log and commented out parts to remove.

**Review by chin9 — APPROVED** [review]
besides val's comment everything looks good!

**Pull request merged**